app/controllers/idv/in_person/state_id_controller.rb

```diff
@@ -16,6 +16,52 @@ def show
         render :show, locals: extra_view_variables
       end
 
+      def update
+        # don't clear the ssn when updating address, clear after SsnController
+        clear_future_steps_from!(controller: Idv::InPerson::SsnController)
+
+        pii_from_user = flow_session[:pii_from_user]
+        initial_state_of_same_address_as_id = flow_session[:pii_from_user][:same_address_as_id]
+        Idv::StateIdForm::ATTRIBUTES.each do |attr|
+          flow_session[:pii_from_user][attr] = flow_params[attr]
+        end
+        form_result = form.submit(flow_params)
+
+        analytics.idv_in_person_proofing_state_id_submitted(
+          **analytics_arguments.merge(**form_result.to_h),
+        )
+
+        if form_result.success?
+          # Accept Date of Birth from both memorable date and input date components
+          formatted_dob = MemorableDateComponent.extract_date_param flow_params&.[](:dob)
+          pii_from_user[:dob] = formatted_dob if formatted_dob
+
+          if pii_from_user[:same_address_as_id] == 'true'
+            copy_state_id_address_to_residential_address(pii_from_user)
+            flow_session['Idv::Steps::InPerson::AddressStep'] = true
+            redirect_url = idv_in_person_ssn_url
+          end
+
+          if initial_state_of_same_address_as_id == 'true' &&
+             pii_from_user[:same_address_as_id] == 'false'
+            clear_residential_address(pii_from_user)
+          end
+
+          if (idv_session.ssn && pii_from_user[:same_address_as_id] == 'true') ||
+             initial_state_of_same_address_as_id == 'false'
+            redirect_url = idv_in_person_verify_info_url
+          elsif pii_from_user[:same_address_as_id] == 'false'
+            redirect_url = idv_in_person_address_url
+          else
+            redirect_url = idv_in_person_ssn_url
+          end
+
+          redirect_to redirect_url if redirect_url
```

Comment:
Nit: Idk if we need the `if redirect_url` part here, because the else on lines 55..57 guarantee that it'll be defined right?

Reply:
good point